src/cluster.py

```diff
@@ -95,6 +122,62 @@ def _get_postgresql_version(self) -> str:
         # Remove the Ubuntu revision from the version.
         return str(package.version).split("+")[0]
 
+    def get_primary(self, unit_name_pattern=False) -> str:
+        """Get primary instance.
+
+        Args:
+            unit_name_pattern: whether to convert pod name to unit name
+
+        Returns:
+            primary pod or unit name.
+        """
+        primary = None
+        # Request info from cluster endpoint (which returns all members of the cluster).
+        cluster_status = requests.get(f"http://{self.unit_ip}:8008/cluster")
+        for member in cluster_status.json()["members"]:
+            if member["role"] == "leader":
+                primary = member["name"]
+                if unit_name_pattern:
+                    # Change the last dash to / in order to match unit name pattern.
+                    primary = "/".join(primary.rsplit("-", 1))
+                break
+        return primary
```

Comment:
This might be cleaner. But tbh I don't really care, feel free to keep what you have
```suggestion
                    primary = "/".join(primary.rsplit("-", 1))
                    return primary
```

Reply:
It looks cleaner. It just probably wouldn't work when there is no primary (it should happen only if there is a problem in a cluster, like a previous issue that I saw in the k8s charm when two clusters have the same name and all members become replicas). In that case it would return None (the default primary variable value) if the already implemented code is kept.